src/transformState.js

```diff
@@ -6,6 +6,33 @@
  */
 function transformState(state, actions) {
   // write code here
-}
+  const nextActions = [];
+
+  for (const key1 of actions) {
+    switch (key1.type) {
+      case 'addProperties':
+        Object.assign(state, key1.extraData);
```

Comment:
Don't mutate object or arrays - it will cause unexpected results later on. You should make copy using `Object.assign` or `spread operator`. You mutate your `state`, you should make a copy of `state` and mutate a copy. All input parameters shouldn't be changed.